Sql-Assignment.txt

```diff
@@ -0,0 +1,1240 @@
+| 3008 | Cisneros | rifkin  | SanJose | barcelona |
+| 3011 | Cisneros | rifkin  | SanJose | barcelona |
+| 3002 | Pereira  | motika  | Rome    | london    |
++------+----------+---------+---------+-----------+
+
+71) Find all salespeople who have customers with more than one current order. 
+
+SELECT s.SNUM , s.SNAME FROM SalesPeople s, Customers c WHERE s.SNUM = c.SNUM GROUP BY SNUM HAVING count(c.SNUM) > 1;
++------+--------+
+| SNUM | SNAME  |
++------+--------+
+| 1001 | peel   |
+| 1002 | serres |
++------+--------+
+
+72) Write a query that extracts from the customer’s table every customer assigned to a salesperson, who is currently having at least one another customer(besides the customer being selected) with orders in the Orders Table. 
+
+ SELECT SNUM FROM customers  GROUP BY SNUM HAVING count(SNUM)>1;
++------+
+| SNUM |
++------+
+| 1001 |
+| 1002 |
++------+
+
+73) Write a query on the customer’s table that will find the highest rating in each city. Put the output in this form: for the city (city), the highest rating is (rating). 
+
+SELECT  CITY "(city)", max(RATING) "(rating)" FROM Customers GROUP BY CITY;
++---------+----------+
+| (city)  | (rating) |
++---------+----------+
+| London  |      100 |
+| Rome    |      200 |
+| SanJose |      300 |
+| Berlin  |      300 |
++---------+----------+
+
+74) Write a query that will produce the snum values of all salespeople with orders, having amt greater than 1000 in the Orders Table(without repeats). 
+
+ select distinct(salespeople.SNUM) from salespeople, orders where salespeople.SNUM = orders.snum and AMT > 1000 order by snum;
++------+
+| SNUM |
++------+
+| 1001 |
+| 1002 |
+| 1003 |
+| 1004 |
+| 1007 |
++------+
+
+75) Write a query that lists customers in a descending order of rating. Output the rating field first, followed by the customer’s names and numbers.
+
+ select rating, cname, cnum from customers order by rating desc;
++--------+----------+------+
+| rating | cname    | cnum |
++--------+----------+------+
+|    300 | Grass    | 2004 |
+|    300 | Cisneros | 2006 |
+|    200 | Giovanni | 2002 |
+|    200 | Liu      | 2003 |
+|    100 | Hoffman  | 2001 |
+|    100 | Clemens  | 2005 |
+|    100 | Pereira  | 2007 |
++--------+----------+------+
+
+76) Find the average commission for salespeople in London. 
+
+ SELECT AVG(COMM) FROM salespeople;
++-----------+
+| AVG(COMM) |
++-----------+
+|   14.3333 |
++-----------+
+
+77) Find all orders credited to the same salesperson who services Hoffman.(cnum 2001). 
+
+ select c.cname, s.sname, o.onum from customers c, salespeople s, orders o where c.SNUM = s.SNUM and c.CNUM = o.CNUM and c.cnum = 2001;
++---------+-------+------+
+| cname   | sname | onum |
++---------+-------+------+
+| Hoffman | peel  | 3003 |
++---------+-------+------+
+
+78) Find all salespeople whose commission is in between 10 and 12(both inclusive). 
+
+SELECT * FROM roneet.salespeople where COMM between 10 AND 12;
```

Comment:
Have you copied the queries from Roneet's assignment? If yes, then this is bad. I said that you should not do it.